File: 연습장/202308/11_금/26. 피보나치 수.py
"""
* Title: 피보나치 수
https://school.programmers.co.kr/learn/courses/30/lessons/12945
"""
def solution(n):
    answer = []

    for i in range(n + 1):
        if i == 0 or i == 1:
            answer.append(i)
        else:
            fibo = answer[i - 1] + answer[i - 2]
            answer.append(fibo % 1234567)

    return answer[-1]

File: _프로그래머스 js/26. 피보나치 수.js
/*
* Title: 피보나치 수
https://school.programmers.co.kr/learn/courses/30/lessons/12945
*/
const solution = (n) => {
  let ls = [0, 1];

  for (let i = 2; i < n + 1; i++) {
    const fibo = ls[i - 1] + ls[i - 2];
    ls.push(fibo % 1234567);
  }

  return ls.pop();
};

// 5
console.log(solution(5));

// 55
console.log(solution(10));
